script/color-contrast.ts

```diff
@@ -0,0 +1,173 @@
+  if (colorAlpha === 1) {
+    return color
+  }
+  // parse BG
+  const [bgR, bgG, bgB, bgAlpha] = parseToRgba(background)
+  // return mixed color
+  const mixed = normal({ r: bgR, g: bgG, b: bgB, a: bgAlpha }, { r: colorR, g: colorG, b: colorB, a: colorAlpha })
+  return rgba(mixed.r, mixed.g, mixed.b, mixed.a)
+}
+/**
+ * runContrastTest
+ * @description runs through all color pairs of a theme and checks the contrasts
+ * @param colorPairs 
+ * @param colors 
+ * @returns contrastTestResult
+ */
+const runContrastTest = (colorPairs: ContrastRequirement[], colors: any): contrastTestResult[] =>
+  // Object.fromEntries(
+  colorPairs.flatMap(([minimumContrast, colorA, colorB, options]: ContrastRequirement) => {
+    // concat name
+    const contrastPair = `${colorA} vs. ${colorB}`
+    // build required string
+    const minimumContrastRatio = `${minimumContrast}:1`;
+    // colorB is fully opaque
+    if (parseToRgba(colors[colorB])[3] === 1) {
+      return {
+        contrastPair,
+        ...testContrast(minimumContrast, colors[colorA], colors[colorB], undefined, contrastPair),
+        minimumContrastRatio
+      }
+    }
+    // if colorB is semi-transparent 
+    // get the correct canvas colors to test agains
+    let canvasColorArrays = cavnasColors
+    // overwrite background if custom canvas array is set
+    if (options?.canvas) canvasColorArrays = options.canvas
+    // test transparent colorB with default bgs `cavnasColors`
+    return canvasColorArrays.map(bg => ({
+      contrastPair: `${contrastPair} on ${bg}`,
+      ...testContrast(minimumContrast, colors[colorA], colors[colorB], colors[bg], contrastPair),
+      minimumContrastRatio
+    })
+    )
+  })
+// )
+/**
+ * testContrast
+ * @description test the contrast of two colors against each other
+ * @param minimumContrast 
+ * @param colorA 
+ * @param colorB 
+ * @param bg used to calculate an opaque color if colorB is semi transparent
+ * @param contrastPair use for better error messages
+ * @returns 
+ */
+const testContrast = (minimumContrast: number, colorA: string, colorB: string, bg: string = '#ffffff', contrastPair?: string): { pass: string, contrastRatio: string } => {
+  // get contrast
+  let contrast = 0
+  try {
+    colorB = getOpaqueColor(colorB, bg)
+    colorA = getOpaqueColor(colorA, colorB)
+    // get contrast rounded down with 2 decimals
+    contrast = Math.floor(getContrast(colorA, colorB) * 100) / 100
+  } catch (err) {
+    console.error(`${contrastPair || ""} as ${colorA} vs.${colorB}: ${err}`)
+  }
+  return {
+    pass: contrast >= minimumContrast ? '✅' : '❌',
+    contrastRatio: `${contrast}: 1`
+  }
+}
+/**
+ * renderConsoleTable
+ * @description takes the test results per theme and prints a nicely formatted table of the results to the console 
+ * @param theme 
+ * @param results 
+ */
+const renderConsoleTable = (theme: string, results: contrastTestResult[]): void => {
+  // config table
+  const contrastTable = new Table({
```

Comment:
Any chance we can do this without the dependency? Appreciate it's the quicker option, but it leaves us open to security and maintenance burden over time that becomes difficult to justify for otherwise straightfoward tasks. Happy to help rewrite this in a part 2 PR if that helps too!

Reply:
Yeah, lets book this for a 2nd PR please.

Reasons for the dep:
- Table looks better
- center aligned columns with `console.table` are hard to read

Reply:
>Table looks better 

FYI.. worth bearing in mind that comments in the Pull Request will strip out all extra styling

Reply:
But not when using it locally.